core/src/Revolution/Processors/Browser/File/Download.php

```diff
@@ -40,7 +40,11 @@ public function process()
 
         // Manager asks for file url
         if (!$this->getProperty('download')) {
-            return $this->success('', ['url' => rawurlencode($this->source->getObjectUrl($file))]);
+            if ($this->source->isFile($file)) {
+                return $this->success('', ['url' => rawurlencode($file)]);
+            } else {
+                return $this->failure($this->modx->lexicon('file_err_nf') . ': ' . $file);
```

Comment:
The lex entry `file_err_nf` has punctuation at the end of if, so it looks strange adding the colon. A couple of suggested options:
1. Change this line to `return $this->failure($this->modx->lexicon('file_err_nf') . ' (' . $file . ')');` OR
2. Consider adding a new file lex entry to provide a smoother, more descriptive message:
`$_lang['file_err_notfound'] = 'The requested file “[[+file]]” could not be found. Please refresh this page to ensure its file listings are up to date.';`
and adjust this line to:
`return $this->failure($this->modx->lexicon('file_err_notfound', ['file' => $file]));`

Reply:
The other messages (i.e. deleting a not existing file) are created the same. That can be changed later on.